packages/web3-eth/test/integration/subscription_new_pending_tx.test.ts

```diff
@@ -35,49 +34,82 @@ describeIf(isWs)('subscription', () => {
 	let web3Eth: Web3Eth;
 	let providerWs: WebSocketProvider;
 	let clientUrl: string;
-	let accounts: string[] = [];
-	beforeAll(async () => {
+	beforeAll(() => {
 		clientUrl = getSystemTestProvider();
-		accounts = await getSystemTestAccounts();
-		providerWs = new WebSocketProvider(
-			clientUrl,
-			{},
-			{ delay: 1, autoReconnect: false, maxAttempts: 1 },
-		);
+		providerWs = new WebSocketProvider(clientUrl);
 	});
 	afterAll(() => {
 		providerWs.disconnect();
 	});
 
 	describe('new pending transaction', () => {
-		it.each(subNames)(`wait ${checkTxCount} transaction - ${subNames[0]}`, async subName => {
+		it.each(subNames)(`wait ${checkTxCount} transaction - %s`, async subName => {
+			const [tempAcc, tempAcc2] = await Promise.all([
+				createTempAccount(),
+				createTempAccount(),
+			]);
 			web3Eth = new Web3Eth(providerWs as Web3BaseProvider);
 			const sub: NewPendingTransactionsSubscription = await web3Eth.subscribe(subName);
-			const from = accounts[0];
-			const to = accounts[1];
+			const from = tempAcc.address;
+			const to = tempAcc2.address;
 			const value = `0x1`;
 
 			let times = 0;
 			const txHashes: string[] = [];
-			const pr = new Promise((resolve: Resolve) => {
-				sub.on('data', (data: string) => {
-					txHashes.push(data);
-					times += 1;
+			let receipts: string[] = [];
+
+			const pr = new Promise((resolve: (s?: string) => void) => {
+				(async () => {
+					let waitList: string[] = [];
+					sub.on('data', (data: string) => {
+						if (receipts.length > 0 && waitList.length > 0) {
+							for (const hash of waitList) {
+								if (receipts.includes(hash)) {
+									txHashes.push(hash);
+									times += 1;
+								}
+							}
+							waitList = [];
+						}
+						if (receipts.length > 0 && receipts.includes(data)) {
+							txHashes.push(data);
+							times += 1;
+						} else {
+							waitList.push(data);
+						}
+
+						if (times >= checkTxCount) {
+							resolve();
+						}
+					});
+					receipts = (
+						await sendFewTxes({
+							web3Eth,
+							from,
+							to,
+							value,
+							times: checkTxCount,
+						})
+					).map(r => String(r?.transactionHash));
+					if (receipts.length > 0 && waitList.length > 0) {
+						for (const hash of waitList) {
+							if (receipts.includes(hash)) {
+								txHashes.push(hash);
+								times += 1;
+							}
+						}
+						waitList = [];
+					}
 					if (times >= checkTxCount) {
 						resolve();
 					}
-				});
-			});
-
-			const receipts = await sendFewTxes({
-				web3Eth,
-				from,
-				to,
-				value,
-				times: checkTxCount,
+				})().catch(console.error);
```

Comment:
It is dangerous to just log the error without firing it. If there was an error at the start of his promise, then both `txHashes` and `receipts` would have an equal length or they can both be just empty. And this means that the next `for` can falsely not detect any possible issue.